src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,45 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+
+        try {
+            Rectangle rectangle = (Rectangle)obj;
+            if (color == null) {
+                if (rectangle.color != null) {
+                    return false;
+                }
+            }
+            if (obj.getClass().equals(Rectangle.class)) {
+                return ((color == null && rectangle.color == null)
+                        || rectangle.getWidth().equals(getWidth()))
+                        && ((width == null && rectangle.width == null)
+                        || rectangle.getLength().equals(getLength()))
+                        && ((length == null && rectangle.length == null)
+                        || rectangle.getColor().equals(getColor()));
```

Comment:
let’s follow the following logical flow for each field:

return true,
if both `field` and `rectangle.field` are null
OR if `field` is not null and equals `rectangle.field`

Reply:
done (added additional checking for each fileds if it is not null)